### DISCUSSION TITLE
Styles are missing, but only in production (build seems to succeed)

### DISCUSSION THREAD

**Discussion by iamgoodbytes** [rappasoft/laravel-livewire-tables#1246] [Q&A] [answered]
I've read a couple of other treads around here regarding missing styles but didn't find a solution that worked in my case. I'm using laravel 10 with a vitejs build and everything works fine locally, like so:

<img width="816" alt="Screenshot 2023-06-09 at 13 38 35" src="https://github.com/rappasoft/laravel-livewire-tables/assets/3899360/89b40c39-8786-4d1f-9fa1-619bd66fc9b4">

When I deploy, all styles work except for the datatable styling. 
<img width="803" alt="Screenshot 2023-06-09 at 13 38 38" src="https://github.com/rappasoft/laravel-livewire-tables/assets/3899360/342a43f1-640c-41e6-bddc-fba4317046f4">

My deployment triggers the vitejs build, so that seemed to have worked fine. 

<img width="724" alt="Screenshot 2023-06-09 at 13 49 59" src="https://github.com/rappasoft/laravel-livewire-tables/assets/3899360/b6353340-9d8a-4ae7-90ec-87d7eb93c30a">

If anyone has an idea on where to look for these missing styles please let me know. 😎

**Comment by lrljoe** [comment, marked as answer]
Make sure you're including the Rappasoft path in your Tailwind config!

See the line in the content[]

Without that, Tailwind won't include the relevant classes, you'll need to do the same for any other packages that include views that are not published!

```
const defaultTheme = require('tailwindcss/defaultTheme');

/** @type {import('tailwindcss').Config} */
module.exports = {
    plugins: [require('@tailwindcss/forms')],
    content: [
        './vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php',
        './vendor/rappasoft/laravel-livewire-tables/resources/views/**/*.blade.php',
        './storage/framework/views/*.php',
        './resources/views/**/*.blade.php',
    ],
   darkMode: 'class',
    theme: {
        extend: {
            fontFamily: {
                sans: ['Figtree', ...defaultTheme.fontFamily.sans],
            },
        },
    },
    variants: {
        extend: {
            backgroundColor: ['responsive', 'dark', 'checked', 'disabled', 'hover', 'focus', 'active', 'even', 'odd'],
            opacity: ['dark'],
            overflow: ['hover'],
        },
    },
};
```

**Comment by iamgoodbytes** [comment]
@lrljoe makes sense and it's indeed right in the docs, just didn't need it locally for some reason even when doing the same build? What would cause the difference between building locally or online? 

Btw: it's fixed, just trying to understand why this happens only in production. Is it an env setting for tailwind or vite?

**Reply by lrljoe** [reply]
In Dev, you're almost certainly running vite in dev mode/just in time (if you're not then you're missing a trick!).  With that it generates the CSS files on the fly based on what's on the page, which gives a slight delay, but means you can tinker more easily.
Great for development, awful for production caching.

npm run build/production optimises all of the files to strip out anything that isn't needed to reduce file sizes.  When you call the tailwind function, it does the same thing, and only adds the classes that it knows about into your compiled CSS file.

There's a lot that can be adjusted for tailwind/vite production builds, if you paste both your tailwind.config.js and vite.config.js here then I'll give you a couple of pointers if I spot anything obvious;
